### ISSUE TITLE
[SSM/GDN] Handle attention_mask in GatedDeltaNet forward pass

### ISSUE THREAD

**Issue by dylanyunlon** [dylanyunlon/Neuron_SP#180]
## 问题描述

`GatedDeltaNet.forward()` 接受 `attention_mask` 参数但完全忽略它，注释 `# TODO: Deal with attention_mask`。

## 代码位置

**文件**: `deepspeed/core/ssm/gated_delta_net.py`  
**行号**: 296

```python
# TODO: Deal with attention_mask
inference_context = deprecate_inference_params(inference_context, inference_params)
seq_len, batch, _ = hidden_states.shape
```

## Megatron 参照

Mamba 模型在处理 packed sequences（THD format）时通过 `packed_seq_params` 传递 attention mask 信息。GDN 需要类似处理来正确处理 padding。

## 建议实现方案

1. 对于 causal SSM，attention_mask 主要用于 masking padding tokens
2. 在 SSM scan 前用 mask zero-out padding positions 的输入
3. 或在输出 after scan 时 mask 掉 padding 位置
4. 与 `packed_seq_params.qkv_format == 'thd'` 分支统一处理

## 影响范围

- Variable-length sequence training（batch packing）
- Padding 处理正确性
